File: src/main/java/org/kite9/diagram/common/algorithms/fg/Arc.java
package org.kite9.diagram.common.algorithms.fg;

public interface Arc {

	public abstract int getFlowFrom(Node n);

	public abstract void pushFlow(int flow);

	public abstract Node getFrom();

	public abstract Node getTo();
	
	public void setFrom(Node from);

	public void setTo(Node to);
	
	public int getIncrementalCost(int flow);
	
	public int getFlow();
	
	public void setFlow(int flow);
	
	public Node otherEnd(Node n);
	
	public void setBlocked(boolean blocked);
	
	public boolean isBlocked();
	
	public boolean hasCapacity(boolean reversed);

}

File: kite9-visualization/src/commonMain/kotlin/org/kite9/diagram/common/algorithms/fg/Arc.kt
package org.kite9.diagram.common.algorithms.fg

import org.kite9.diagram.common.algorithms.det.Deterministic

interface Arc : Deterministic {

    fun getFlowFrom(n: Node): Int
    fun pushFlow(flow: Int)
    var from: Node
    var to: Node
    fun getIncrementalCost(flow: Int): Int
    var flow: Int
    fun otherEnd(n: Node): Node
    fun hasCapacity(reversed: Boolean): Boolean
}